File: 0334-increasing-triplet-subsequence/0334-increasing-triplet-subsequence.cpp
class Solution {
public:
    bool increasingTriplet(vector<int>& nums) {
        int first = INT_MAX; 
        int second =  INT_MAX; 
        for(int i = 0; i < nums.size(); i++){
            int curr = nums[i]; 
            if(first >= curr){
                first = curr; 
            } else if(second >= curr){
                second = curr; 
            } else{
                return true; 
            }
        }

        return false; 
    }
};

File: 0334-increasing-triplet-subsequence/0334-increasing-triplet-subsequence.java
class Solution {
    public boolean increasingTriplet(int[] nums) {
        int first = nums[0]; 
        int second = Integer.MAX_VALUE; 
        int third = Integer.MAX_VALUE; 
        for(int i = 1; i < nums.length; i++){
            if(nums[i] <= first){
                first = nums[i]; 
            } else if(nums[i] <= second){
                second = nums[i]; 
            } else{
                return true; 
            }
        }

        return false; 
    }
}